book/src/storage.md

```diff
@@ -0,0 +1,52 @@
+# Persistent Storage
+
+Ariel OS supports persistent storage on a number of boards.
+It provides a key–value pair store based on [sequential-storage].
+The storage sets up the flash storage required.
+
+## Use Cases
+
+Persistent storage can be used to store configuration and other variables
+across reboots and power interruptions on the microcontroller.
+
+## Usage
+
+### Enabling the Storage Module
+
+The storage module can be enabled by [selecting the `sw/storage` laze module][laze-modules-book].
+
+### Using the Storage Module
+
+The [storage module] provides the necessary getters and setters
+to interact with the storage backend.
+
+Keys are always `&str` typed.
+Values are required to implement the [`serde::Serialize`][serde-serialize]
+and [`serde::Deserialize`][serde-deserialize] traits.
+Care must be taken to always use the same value type with the same key.
```

Comment:
```suggestion
Care must be taken to always read a key with the same value type that it was written before.
While using a different value type for reading than for writing is never unsafe, it might result in bogus data.
```